src/components/movies/MovieList.tsx

```diff
@@ -0,0 +1,204 @@
+//auth
+import { useAuth } from '../auth/authContext';
+
+//Component
+import Banner from '../banner';
+
+//Services
+import { allMovies, moviesUser, deleteMovieUser, addingMovie } from '../../services';
+
+//CSS
+import '../../styles/brower.scss';
+
+//type
+import { PageData } from '../../types';
+
+//utils
+import { formatDate } from '../../utils/formatters';
+
+/***Framework***/
+//Fontawesome
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faHeart as faHeartSolid } from '@fortawesome/free-solid-svg-icons';
+import { faHeart as faHeartRegular } from '@fortawesome/free-regular-svg-icons';
+//MUI
+import IconButton from '@mui/material/IconButton';
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import CardHeader from '@mui/material/CardHeader';
+import Typography from '@mui/material/Typography';
+import CardMedia from '@mui/material/CardMedia';
+import Pagination from '@mui/material/Pagination';
+import Stack from '@mui/material/Stack';
+
+//React
+import { useState } from 'react';
+import useSWR, { mutate } from 'swr';
+
+function Browser() {
+  //valeur de la page
+  const [page, setPage] = useState<number>(1);
+  //auth
+  const { user } = useAuth();
+
+  //useSWR qui gère la récuparation de tout les film, la connection et si connecté les favoris de l'utilisateur
+  const { data, isLoading } = useSWR<PageData>('pageData', async () => {
+    //PageData regroupe 2 type l'ApiMovies et Usermovie
+    const movies = await allMovies(); // tous les films
+
+    let userMovies = [];
+    if (user?.id) {
+      // si connecté
+      userMovies = await moviesUser(user.id); //liste des favoris de l'utilisateur
+    }
+
+    return {
+      //data 2 valeurs userMovies peut-être vide
+      movies,
+      userMovies,
+    };
+  });
+
+  if (isLoading) return <p>Loading...</p>;
+  if (!data) return <p>Error</p>;
+
+  const movies = data.movies;
```

Comment:
Si tu veux tu peux écrire ça
```suggestion
  const {
      movies,
      userMovies
  } = data;
```